### PR TITLE
#5 user details page - simplest implementation

### PR DESCRIPTION
**Pull request by maraspin**

### CODE REVIEW

File: app/config/routes.php [hunk 1]
```diff
@@ -40,7 +40,13 @@
 
 $app->post('/login', \App\Action\LoginAction::class, 'login');
 
-$app->get('/users', \App\Action\UsersAction::class, 'users');
+$app->get('/users', \App\Action\UserAction::class, 'users');
```
Comment (marcosh) [inline app/config/routes.php]: why did you change this name?

File: app/src/App/Domain/Service/UsersService.php [hunk 2]
```diff
@@ -36,7 +36,13 @@ public function getAll(): array
 
     public function getById(UserId $id): ?User
```
Comment (marcosh) [inline app/src/App/Domain/Service/UsersService.php]: this does not return `null` now. Please remove the `?`

File: app/src/App/Domain/Service/UsersService.php [hunk 3]
```diff
@@ -36,7 +36,13 @@ public function getAll(): array
...
     {
-        return $this->users->findById($id);
+        $user = $this->users->findById($id);
+
+        if (!$user instanceof User) {
+            throw UserNotFoundException::fromUsername($username);
```
Comment (marcosh) [inline app/src/App/Domain/Service/UsersService.php]: `$username` is not defined here

Comment (marcosh) [inline app/src/App/Domain/Service/UsersService.php]: since now we're throwing an exception, it should be the case to add a test for it

File: app/config/routes.php (hunk 1 above)
Comment (maraspin) [reply app/config/routes.php]: Users Actions sounded strange to me. Just found out (http://www.learnersdictionary.com/qa/can-a-plural-noun-act-as-an-adjective) it's a very personal rather than grammatical thing. Also, I took a chance to change something and see how that would impact FW workflow. Can revert if necessary.

File: app/src/App/Domain/Service/UsersService.php (hunk 2 above)
Comment (maraspin) [reply app/src/App/Domain/Service/UsersService.php]: Too much fun coding again. My fault.

File: app/src/App/Domain/Service/UsersService.php (hunk 3 above)
Comment (maraspin) [reply app/src/App/Domain/Service/UsersService.php]: As comment before... will fix.

Comment (maraspin) [reply app/src/App/Domain/Service/UsersService.php]: Yep, I was just about to make a new commit with it ...been hit too many times in the past to ignore https://en.wikipedia.org/wiki/Confirmation_bias :-)

File: app/config/routes.php (hunk 1 above)
Comment (marcosh) [reply app/config/routes.php]: please do, or change also the `ChocolatesAction`

File: app/src/App/Domain/Service/UsersService.php [hunk 4]
```diff
@@ -34,15 +34,21 @@ public function getAll(): array
         return $this->users->all();
     }
 
-    public function byId(UserId $id): ?User
+    public function getById(UserId $id): ?User
```
Comment (marcosh) [inline app/src/App/Domain/Service/UsersService.php]: these methods are surely used around in the code. Please rename them everywhere

**Review by marcosh — COMMENTED** [review]
sddf

**Comment by maraspin** [comment]
Issues above should be all addressed now
